src/reactors/V2DutchOrderReactor.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {BaseReactor} from "./BaseReactor.sol";
+import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
+import {Permit2Lib} from "../lib/Permit2Lib.sol";
+import {DutchDecayLib} from "../lib/DutchDecayLib.sol";
+import {V2DutchOrderLib, V2DutchOrder, CosignerData, DutchOutput, DutchInput} from "../lib/V2DutchOrderLib.sol";
+import {SignedOrder, ResolvedOrder} from "../base/ReactorStructs.sol";
+
+/// @notice Reactor for v2 dutch orders
+/// @dev V2 orders must be cosigned by the specified cosigner to override timings and starting values
+/// @dev resolution behavior:
+/// - If cosignature is invalid or not from specified cosigner, revert
+/// - If inputOverride is 0, then use inner inputs
+/// - If inputOverride is nonzero, then ensure it is less than specified input and replace startAmount
+/// - For each DutchOutput:
+///   - If override is 0, then use inner output
+///   - If override is nonzero, then ensure it is greater than specified output and replace startAmount
+contract V2DutchOrderReactor is BaseReactor {
+    using Permit2Lib for ResolvedOrder;
+    using V2DutchOrderLib for V2DutchOrder;
+    using DutchDecayLib for DutchOutput[];
+    using DutchDecayLib for DutchInput;
+
+    /// @notice thrown when an order's deadline is before its end time
+    error DeadlineBeforeEndTime();
+
+    /// @notice thrown when an order's inputs and outputs both decay
+    error InputAndOutputDecay();
+
+    /// @notice thrown when an order's cosignature does not match the expected cosigner
+    error InvalidCosignature();
+
+    /// @notice thrown when an order's input override is greater than the specified
+    error InvalidInputOverride();
+
+    /// @notice thrown when an order's output override is less than the specified
+    error InvalidOutputOverride();
+
+    constructor(IPermit2 _permit2, address _protocolFeeOwner) BaseReactor(_permit2, _protocolFeeOwner) {}
+
+    /// @inheritdoc BaseReactor
+    function resolve(SignedOrder calldata signedOrder)
+        internal
+        view
+        virtual
+        override
+        returns (ResolvedOrder memory resolvedOrder)
+    {
+        V2DutchOrder memory order = abi.decode(signedOrder.order, (V2DutchOrder));
+        // hash the order _before_ overriding amounts, as this is the hash the user would have signed
+        bytes32 orderHash = order.hash();
+
+        _validateOrder(orderHash, order);
+        _updateWithOverrides(order);
+
+        resolvedOrder = ResolvedOrder({
+            info: order.info,
+            input: order.input.decay(order.cosignerData.decayStartTime, order.cosignerData.decayEndTime),
```

Comment:
Maybe I've misunderstood... but it feels like a strange UX that if a cosigner is overriding, they dont override with the actual amount that they want to fill at, but instead they override a new `startAmount` and have to predict what time their tx will be mined to get the correct `decay` to the amount they actually want to pay?
It would feel more natural to me that onchain it should calculate the `decay` and then call `_updateWithOverrides` above the _decayed_ value?